libsolidity/analysis/TypeChecker.cpp

```diff
@@ -4041,12 +4041,19 @@ void TypeChecker::endVisit(Literal const& _literal)
 					// This was already validated in visit(Literal) but the error is not fatal.
 					if (!mantissa || !exponent)
 						solAssert(!m_errorReporter.errors().empty());
-					else if (
-						!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)) ||
-						!exponent->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(1))
-					)
-						// TODO: Is this triggered when the argument is out of range? Test.
-						parameterTypeMessage = "The type of the literal cannot be converted to the parameters of the suffix function.";
+					// TODO: Is this triggered when the argument is out of range? Test.
+					else if (!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)))
+						parameterTypeMessage = fmt::format(
+							"Type {} of the mantissa cannot be converted to type {} of the suffix function parameter.",
+							mantissa->mobileType()->toString(true),
+							suffixFunctionType->parameterTypes().at(0)->toString(true)
+						);
+					else if (!exponent->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(1)))
+						parameterTypeMessage = fmt::format(
+							"Type {} of the exponent cannot be converted to type {} of the suffix function parameter.",
+							exponent->mobileType()->toString(true),
+							suffixFunctionType->parameterTypes().at(1)->toString(true)
+						);
 				}
 			}
 			else if (suffixFunctionType->parameterTypes().size() == 1)
```

Comment:
I am working on changing the message inside this block as well, but I am getting a segfault which I haven't figured out the solution (or why it happens) yet. It occurs when I try to get the name of the type using `literalRationalType->mobileType()->toString(true)`